pkg/vm/vm.go

```diff
@@ -376,7 +376,7 @@ func (v *VM) StepOut() error {
 	}
 
 	expSize := v.istack.len
-	for v.state.HasFlag(noneState) && v.istack.len >= expSize {
+	for v.state == noneState && v.istack.len >= expSize {
```

Comment:
I think vm changes are worth an additional commit that would explain changes done.

Reply:
It is here to match original VM behavior. I am not sure if there is a bug in our or their implementation.

Reply:
I think it's a bug or ours because `noneState` is not a flag.